### PR TITLE
multi: fix loopvar and enhance logging around channel reestablishment

### PR DESCRIPTION
**Pull request by yyforyongyu**
Attempt to provide more logs so we can find out what's the root cause in #8130

### CODE REVIEW

**Review by bhandras — APPROVED** [review]
LGTM 🌮

File: channeldb/channel.go [hunk 1]
```diff
@@ -849,6 +849,25 @@ type OpenChannel struct {
...
+func (c *OpenChannel) String() string {
+	indexStr := "height=%v, htlc_index=%v, log_index=%v"
+
+	commit := c.LocalCommitment
+	local := fmt.Sprintf(indexStr, commit.CommitHeight,
+		commit.LocalHtlcIndex, commit.LocalLogIndex)
+
+	commit = c.RemoteCommitment
+	remote := fmt.Sprintf(indexStr, commit.CommitHeight,
+		commit.LocalHtlcIndex, commit.LocalLogIndex)
```
Comment (ellemouton) [inline channeldb/channel.go:862]: meant to be `commit.RemoteHtlcIndex, commit.RemoteLogIndex)` right?

File: channeldb/channel.go [hunk 2]
```diff
@@ -2108,8 +2131,16 @@ func (c *OpenChannel) ActiveHtlcs() []HTLC {
...
 	activeHtlcs := make([]HTLC, 0, len(remoteHtlcs))
 	for _, htlc := range c.LocalCommitment.Htlcs {
+		log.Debugf("LocalCommitment has htlc: id=%v, update=%v "+
+			"incoming=%v", htlc.HtlcIndex, htlc.LogIndex,
+			htlc.Incoming)
+
 		onionHash := sha256.Sum256(htlc.OnionBlob[:])
 		if _, ok := remoteHtlcs[onionHash]; !ok {
+			log.Debugf("Skipped htlc due to onion has not "+
+				"matched: id=%v, update=%v incoming=%v",
```
Comment (ellemouton) [inline channeldb/channel.go:2141]: strange wording. Perhaps "Skipped htlc since the onion does not match" or "Skipped htlc due to onion mismatch"?

**Review by ellemouton — COMMENTED** [review]
Looks good!

File: channeldb/channel.go (hunk 1 above)
Comment (bhandras) [reply channeldb/channel.go:862]: It's addressing the remoteCommitment (just tricks the eye).

Comment (ellemouton) [reply channeldb/channel.go:862]: ah indeed! thanks 🙈

**Review by ellemouton — APPROVED** [review]
LGTM! 🔥

File: channeldb/channel.go (hunk 2 above)
Comment (yyforyongyu) [reply channeldb/channel.go:2141]: fixed!

File: channeldb/channel.go (hunk 1 above)
Comment (yyforyongyu) [reply channeldb/channel.go]: decided to populate both local/remote indexes on both local/remote commitments, and yeah it's tricky😂

**Review by ziggie1984 — APPROVED** [review]
Thanks for this fix, LGTM ⭐️

**Review by Crypt-iQ — APPROVED** [review]
nice catch 🎣

File: channeldb/channel.go [hunk 3]
```diff
@@ -2108,8 +2136,16 @@ func (c *OpenChannel) ActiveHtlcs() []HTLC {
 	// as active if *we* know them as well.
 	activeHtlcs := make([]HTLC, 0, len(remoteHtlcs))
 	for _, htlc := range c.LocalCommitment.Htlcs {
+		log.Debugf("LocalCommitment has htlc: id=%v, update=%v "+
+			"incoming=%v", htlc.HtlcIndex, htlc.LogIndex,
+			htlc.Incoming)
+
 		onionHash := sha256.Sum256(htlc.OnionBlob[:])
 		if _, ok := remoteHtlcs[onionHash]; !ok {
+			log.Debugf("Skipped htlc due to onion mismatched: "+
```
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply channeldb/channel.go:2145]: yeah downgraded to trace instead

**Review by Roasbeef — CHANGES_REQUESTED** [review]

**Comment by Roasbeef** [comment]
Needs a rebase! (release notes conflict)

**Pull request merged**